linkerd/proxy/spire-client/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+mod api;
+
+pub use api::{Api, SvidUpdate};
+use linkerd_error::Error;
+use linkerd_identity::Credentials;
+use linkerd_identity::Id;
+use std::fmt::{Debug, Display};
+use tokio::sync::watch;
+use tower::{util::ServiceExt, Service};
+use tracing::error;
+
+pub struct Spire {
+    id: Id,
+}
+
+// === impl Spire ===
+
+impl Spire {
+    pub fn new(id: Id) -> Self {
+        Self { id }
+    }
+
+    pub async fn run<C, S>(self, credentials: C, mut client: S)
+    where
+        C: Credentials,
+        S: Service<(), Response = tonic::Response<watch::Receiver<SvidUpdate>>>,
+        S::Error: Into<Error> + Display + Debug,
+    {
+        let client = client.ready().await.expect("should be ready");
+        match client.call(()).await {
+            Ok(rsp) => consume_updates(&self.id, rsp.into_inner(), credentials).await,
+            Err(error) => error!(%error, "could not establish SVID stream"),
+        }
```

Comment:
```suggestion
        let client = client.ready().await.expect("should be ready");
        let rsp = client.call(()).await.expect("spire client must gracefully handle errors");
        consume_updates(&self.id, rsp.into_inner(), credentials).await
```

Based on our earlier convo, I'm pretty sure that a failure here indicates something pathological (i.e., a bad request from the client).  If so, it seems better to panic than to log and never satisfy an identity.